src/physics/em/detail/RelativisticBremInteractor.i.hh

```diff
@@ -0,0 +1,124 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file RelativisticBremInteractor.i.hh
+//---------------------------------------------------------------------------//
+
+#include "base/ArrayUtils.hh"
+#include "base/Algorithms.hh"
+#include "random/distributions/GenerateCanonical.hh"
+#include "random/distributions/UniformRealDistribution.hh"
+#include "TsaiUrbanDistribution.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with shared and state data.
+ */
+CELER_FUNCTION
+RelativisticBremInteractor::RelativisticBremInteractor(
+    const RelativisticBremNativeRef& shared,
+    const ParticleTrackView&         particle,
+    const Real3&                     direction,
+    const CutoffView&                cutoffs,
+    StackAllocator<Secondary>&       allocate,
+    const MaterialView&              material,
+    const ElementComponentId&        elcomp_id)
+    : shared_(shared)
+    , inc_energy_(particle.energy())
+    , inc_direction_(direction)
+    , gamma_cutoff_(cutoffs.energy(shared.ids.gamma))
+    , allocate_(allocate)
+    , dxsec_(shared, particle, material, elcomp_id)
+{
+    CELER_EXPECT(particle.particle_id() == shared_.ids.electron
+                 || particle.particle_id() == shared_.ids.positron);
+    CELER_EXPECT(gamma_cutoff_.value() > 0);
+
+    // Valid energy region of the relativistic e-/e+ Bremsstrahlung model
+    CELER_EXPECT(inc_energy_ > shared_.low_energy_limit());
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the production of photons using the G4eBremsstrahlungRelModel
+ * of Geant4 6.10.
+ */
+template<class Engine>
+CELER_FUNCTION Interaction RelativisticBremInteractor::operator()(Engine& rng)
+{
+    // Min and max kinetic energy limits for sampling the secondary photon
+    Energy tmin = min(gamma_cutoff_, inc_energy_);
+    Energy tmax = min(shared_.high_energy_limit(), inc_energy_);
+    if (tmin >= tmax)
+    {
+        return Interaction::from_unchanged(inc_energy_, inc_direction_);
+    }
+
+    real_type density_corr = dxsec_.density_correction();
+    real_type xmin         = std::log(ipow<2>(tmin.value()) + density_corr);
+    real_type xrange = std::log(ipow<2>(tmax.value()) + density_corr) - xmin;
+
+    real_type gamma_energy{0};
+    real_type dsigma{0};
+
+    do
+    {
+        gamma_energy = std::sqrt(max(
+            0.0,
+            std::exp(xmin + generate_canonical(rng) * xrange) - density_corr));
+        dsigma       = dxsec_(gamma_energy);
+    } while (dsigma < dxsec_.maximum_value() * generate_canonical(rng));
+
```

Comment:
I see that in `G4eBremsstrahlungRelModel::SampleSecondaries` there is an extra piece of logic for handling scattering off of the nucleus or e- by triplet model. Do we want to note the difference here? (not sure under what conditions that's relevant)

Reply:
The triple model is never used in the standard EM physics and the related place-holding API block is actually never called (so useless) in the current Geant4 code.  I even did not bother to put any comment for that.